### ISSUE TITLE
German umlauts (Umlaute öäüÖÄÜß)

### ISSUE THREAD

**Issue by merlinuwe** [twibiral/obsidian-execute-code#200]
Thanks for your great plugin,

Do you know how to get german umlauts (Umlaute: öäüÖÄÜß) as an output?

I often use Python, Powershell and Batch.

![image](https://user-images.githubusercontent.com/4138869/210384186-95d1eba3-10a6-4688-8ed0-3466355009e2.png)

**Comment by chlohal** [comment]
I see, thanks for that breakdown link. That's very helpful!

The plugin uses `utf-8`; could you try using the `$PSDefaultParameterValues['Out-File:Encoding'] = 'utf8'` command to force your default encoding to be `utf-8`?

**Comment by merlinuwe** [comment]
Thanks for your advice.

When I do so:

```powershell
$PSDefaultParameterValues['Out-File:Encoding'] = 'utf8'
Get-ChildItem P:\Obsidian\ -Recurse | where{$_.Length -eq 0} | Select @{N='EmptyFiles';E={$_.FullName}}
```
I get the same problem as before:

![image](https://user-images.githubusercontent.com/4138869/212545864-2d4b74d0-3886-4cc5-8d18-058c50e44be0.png)

There is no difference, when I set your command in Powershell first.

I also tested this

![image](https://user-images.githubusercontent.com/4138869/212545984-3292d301-54a0-4f21-9e17-4b8fd82120e7.png)

**Comment by merlinuwe** [comment]
But when I put in this code into your settings, it works for every powershell script, which runs in Obsidian:

[Console]::OutputEncoding = [System.Text.Encoding]::UTF8

![image](https://user-images.githubusercontent.com/4138869/212547496-01f73361-2cc2-48b8-ac9a-6619cee090ec.png)

---

This works also:

```powershell
Get-ChildItem P:\Obsidian\ -Recurse | where{$_.Length -eq 0} | Select @{N='EmptyFiles';E={$_.FullName}} | out-file -encoding utf8 "P:\filename.md"
```

**Comment by twibiral** [comment]
Hi @merlinuwe, I think the settings you are using now are still not enough. Even if I add the following in the settings:

```
$OutputEncoding = [System.Text.Encoding]::UTF8;
[Console]::OutputEncoding = [System.Text.Encoding]::UTF8;
[Console]::InputEncoding = [System.Text.Encoding]::UTF8;
```

But äöüß still don't work in some situations:

![grafik](https://user-images.githubusercontent.com/46483257/212671112-3d2ca6b1-ebeb-4a90-b02b-0779eb4a863b.png)

I will add more documentation as soon as we figure the best settings out and just set them as default.

**Comment by twibiral** [comment]
Fix released in [1.7.0](https://github.com/twibiral/obsidian-execute-code/releases/tag/1.7.0).

@merlinuwe You need to set your PowerShell settings to:
```powershell
$OutputEncoding = [console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding
```

The problem was that PowerShell still uses `windows-1252` as the default encoding on windows for legacy reasons. You can set your whole system to use `utf-16` too, but now the plugin assumes by default that `windows-1252` is used.
